crates/ra_ide_api/src/hover.rs

```diff
@@ -86,13 +86,13 @@ pub(crate) fn hover(db: &RootDatabase, position: FilePosition) -> Option<RangeIn
         use crate::goto_definition::{ReferenceResult::*, reference_definition};
         let ref_result = reference_definition(db, position.file_id, name_ref);
         match ref_result {
-            Exact(nav) => res.extend(doc_text_for(db, nav)),
```

Comment:
The next refactoring would be to fix this code to not depend on nav target at all. Instead of `reference_definition`, whihc returns `NavTarget`s, hover should call `classify_name_ref`, which returns hir elements.